### PR TITLE
Python splitting // adds python separators

### PR DESCRIPTION
**Pull request by stuartjohnpage**
connects to #6 

### Overview

- Adds separators for python, based on [Langchain](https://github.com/langchain-ai/langchainjs/blob/28ff29fc73401d1ae3e9772f7b89089bdd71b282/langchain/src/text_splitter.ts#L482) as an example
- Adds tests for python splitting
- Adds tests for javascript splitting

### CODE REVIEW

**Review by lukeledet — APPROVED** [review]
👍 👍

File: test/support/fixtures/document_fixtures/test_code.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          return true;
+      } else {
+          return false;
+      }
+  }
+
+  getPetCount(petType) {
+      return this.inventory[petType] || 0; 
+  }
+}
```
Comment (grossvogel) [inline test/support/fixtures/document_fixtures/test_code.js:27]: no newline?

File: test/support/fixtures/document_fixtures/test_code.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class PetShop:
+    """Represents a pet shop with inventory and sales functionality."""
+
+    def __init__(self, name):
+        self.name = name
+        self.inventory = {}
+
+    def add_pet(self, pet_type, quantity):
+        """Adds a specified quantity of a pet type to the inventory."""
```
Comment (grossvogel) [inline test/support/fixtures/document_fixtures/test_code.py:9]: I guess it's kinda nice for our purposes that python doc blocks (or whatever they call them) are _inside_ the methods 🤔

**Review by grossvogel — APPROVED** [review]

File: test/support/fixtures/document_fixtures/test_code.py (hunk 2 above)
Comment (stuartjohnpage) [reply test/support/fixtures/document_fixtures/test_code.py:9]: yeah, it would be a pain otherwise

**Review by Dacello — APPROVED** [review]

**Pull request merged**